File: kata-solutions/tell-dont-ask/tell-dont-ask-java/app/src/main/java/useCase/OrderShipmentUseCase.java
package useCase;

import domain.Order;
import repository.OrderRepository;
import service.ShipmentService;

public class OrderShipmentUseCase {
    private final OrderRepository orderRepository;
    private final ShipmentService shipmentService;

    public OrderShipmentUseCase(OrderRepository orderRepository, ShipmentService shipmentService) {
        this.orderRepository = orderRepository;
        this.shipmentService = shipmentService;
    }

    public void run(OrderShipmentRequest request) {
        final Order order = orderRepository.getById(request.getOrderId());

        if (order.canBeShipped()) {
            shipmentService.ship(order);
            order.ship();
            orderRepository.save(order);
        }
    }
}

File: tdd-katas/tell-dont-ask/tell-dont-ask-python/src/useCase/OrderShipmentUseCase.py
from ..domain.OrderStatus import OrderStatus
from ..repository.OrderRepository import OrderRepository
from ..service.ShipmentService import ShipmentService
from ..useCase.OrderCannotBeShippedError import OrderCannotBeShippedError
from ..useCase.OrderCannotBeShippedTwiceError import OrderCannotBeShippedTwiceError
from ..useCase.OrderShipmentRequest import OrderShipmentRequest


class OrderShipmentUseCase(object):
    def __init__(self, order_repository: OrderRepository, shipment_service: ShipmentService):
        self.order_repository = order_repository
        self.shipment_service = shipment_service

    def run(self, request: OrderShipmentRequest):
        order = self.order_repository.get_by_id(request.get_order_id())

        if order.get_status() is OrderStatus.CREATED or order.get_status() is OrderStatus.REJECTED:
            raise OrderCannotBeShippedError()

        if order.get_status() is OrderStatus.SHIPPED:
            raise OrderCannotBeShippedTwiceError()

        self.shipment_service.ship(order)

        order.set_status(OrderStatus.SHIPPED)
        self.order_repository.save(order)
